BBC.js

```diff
@@ -46,20 +46,37 @@ function detectWeb(doc, url) {
 		}
 		return "newspaperArticle";
 	}
-	if(url.indexOf("/newsbeat/article") != -1)
+	if(url.indexOf("/newsbeat/article") != -1){
 		return "blogPost";
+	}
 	if (getSearchResults(doc, true)) {
 		return "multiple";
 	}
 }
 
 function getSearchResults(doc, checkOnly) {
+	var namespace = doc.documentElement.namespaceURI;
+	var nsResolver = namespace ? function(prefix) {
+	if (prefix == 'x') return namespace; else return null;
+	} : null;
+
 	var items = {};
 	var found = false;
 	var rows = ZU.xpath(doc, '//a[h3[@class="title-link__title"]]');
-	for (var i=0; i<rows.length; i++) {
+	//for NewsBeat
```

Comment:
I did this because for newsbeat, rows[i].href wasn't catching the url. Rather there was only one key, baseURL(url of the search page) and no other url information. So to retrieve the url of a specific title, I used this evaluate method. The test case for this is the harry potter search in newbeat.